code/modules/reagents/reagent_containers/food/snacks/grown.dm

```diff
@@ -41,6 +43,206 @@
 		if(reagents.total_volume > 0)
 			bitesize = 1 + round(reagents.total_volume/2, 1)
 
+/obj/item/weapon/reagent_containers/food/snacks/grown/throw_impact(atom/hit_atom)
+	..()
+	if(!seed || !src) return
+	//if(seed.stinging)   			//we do NOT want to transfer reagents on throw, as it would mean plantbags full of throwable chloral injectors
+	//	stinging_apply_reagents(M)  //plus all sorts of nasty stuff like throw_impact not targeting a specific bodypart to check for protection.
+
+	// We ONLY want to apply special effects if we're hitting a turf! That's because throw_impact will always be
+	// called on a turf AFTER it's called on the things ON the turf, and will runtime if the item doesn't exist anymore.
+	if(isturf(hit_atom))
+		do_splat_effects(hit_atom)
+	return
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/proc/do_splat_effects(atom/hit_atom)
+	if(seed.teleporting)
+		splat_reagent_reaction(get_turf(hit_atom))
+		if(do_fruit_teleport(hit_atom, usr, potency))
+			visible_message("<span class='danger'>The [src] splatters, causing a distortion in space-time!</span>")
+		else if(splat_decal(get_turf(hit_atom)))
+			visible_message("<span class='notice'>The [src.name] has been squashed.</span>","<span class='moderate'>You hear a smack.</span>")
+		qdel(src)
+		return
+
+	if(seed.juicy)
+		splat_decal(get_turf(hit_atom))
+		splat_reagent_reaction(get_turf(hit_atom))
+		visible_message("<span class='notice'>The [src.name] has been squashed.</span>","<span class='moderate'>You hear a smack.</span>")
+		qdel(src)
+		return
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/attack(mob/living/M, mob/user, def_zone)
+	if(user.a_intent == I_HURT)
+		. = handle_attack(src,M,user,def_zone)
+		if(seed.stinging)
+			if(M.getarmor(def_zone, "melee") < 5)
+				var/reagentlist = stinging_apply_reagents(M)
+				if(reagentlist)
+					M << "<span class='danger'>You are stung by \the [src]!</span>"
+					add_attacklogs(user, M, "stung", object = src, addition = "Reagents: [english_list(seed.get_reagent_names())]", admin_warn = 1)
+			user << "<span class='alert'>Some of \the [src]'s stingers break off in the hit!</span>"
+			potency -= rand(1,(potency/3)+1)
+		do_splat_effects(M)
+		return
+	return ..()
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/Crossed(var/mob/living/carbon/M)
+	..()
+	if(!seed) return
+	if(!istype(M)) return
+	if(!M.on_foot())
+		return
+	if(seed.thorny || seed.stinging)
+		if(istype(M, /mob/living/carbon/human))
+			var/mob/living/carbon/human/H = M
+			if(!H.check_body_part_coverage(FEET))
+				var/datum/organ/external/affecting = H.get_organ(pick("l_foot", "r_foot"))

```

Comment:
Er, I don't see any sanity against this being null, so if the mob has no legs this will runtime.